py/frequenz/client/electricity_trading/_types.py

```diff
@@ -0,0 +1,1220 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Module to define the types used with the client."""
+
+# pylint: disable=C0302
+
+from __future__ import annotations  # required for constructor type hinting
+
+import enum
+import logging
+from dataclasses import dataclass
+from datetime import datetime
+from decimal import Decimal
+
+from frequenz.api.common.v1.grid import delivery_area_pb2, delivery_duration_pb2
+from frequenz.api.common.v1.market import energy_pb2, price_pb2
+from frequenz.api.common.v1.pagination import pagination_params_pb2
+from frequenz.api.electricity_trading.v1 import electricity_trading_pb2
+from google.protobuf import json_format, struct_pb2, timestamp_pb2
+from google.type import decimal_pb2
+
+_logger = logging.getLogger(__name__)
+
+
+# From frequanz.api.common
+class Currency(enum.Enum):
+    """
+    List of supported currencies.
+
+    New currencies can be added to this enum to support additional markets.
+    """
+
+    UNSPECIFIED = price_pb2.Price.Currency.CURRENCY_UNSPECIFIED
+    """Currency is not specified."""
+
+    USD = price_pb2.Price.Currency.CURRENCY_USD
+
+    CAD = price_pb2.Price.Currency.CURRENCY_CAD
+
+    EUR = price_pb2.Price.Currency.CURRENCY_EUR
+
+    GBP = price_pb2.Price.Currency.CURRENCY_GBP
+
+    CHF = price_pb2.Price.Currency.CURRENCY_CHF
+
+    CNY = price_pb2.Price.Currency.CURRENCY_CNY
+
+    JPY = price_pb2.Price.Currency.CURRENCY_JPY
+
+    AUD = price_pb2.Price.Currency.CURRENCY_AUD
+
+    NZD = price_pb2.Price.Currency.CURRENCY_NZD
+
+    SGD = price_pb2.Price.Currency.CURRENCY_SGD
+
+    @classmethod
+    def from_pb(cls, currency: price_pb2.Price.Currency.ValueType) -> Currency:
```

Comment:
optionally, constructors can annotate returns with: 

```python
from typing import Self
```

```suggestion
    def from_pb(cls, currency: price_pb2.Price.Currency.ValueType) -> Self:
```